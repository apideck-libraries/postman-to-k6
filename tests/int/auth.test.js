/* eslint-disable space-before-function-paren */

import convertFile from '../../lib/convert/file';

test('noauth', async () => {
  const [main] = await convertFile('test/material/2/noauth.json');
  expect(main).toMatchSnapshot();
});

test('basic', async () => {
  const [main] = await convertFile('test/material/2/basic.json');
  expect(main).toMatchSnapshot();
});

test('bearer', async () => {
  const [main] = await convertFile('test/material/2/bearer.json');
  expect(main).toMatchSnapshot();
});

test('digest', async () => {
  const [main] = await convertFile('test/material/2/digest.json');
  expect(main).toMatchSnapshot();
});

test('ntlm', async () => {
  const [main] = await convertFile('test/material/2/ntlm.json');
  expect(main).toMatchSnapshot();
});

test('awsv4', async () => {
  const [main] = await convertFile('test/material/2/awsv4.json');
  expect(main).toMatchSnapshot();
});

test('oauth1 header sha1', async () => {
  const [main] = await convertFile('test/material/2/oauth1-header-sha1.json');
  expect(main).toMatchSnapshot();
});

test('oauth1 header sha256', async () => {
  const [main] = await convertFile('test/material/2/oauth1-header-sha256.json');
  expect(main).toMatchSnapshot();
});

test('oauth1 header text', async () => {
  const [main] = await convertFile('test/material/2/oauth1-header-text.json');
  expect(main).toMatchSnapshot();
});

test('oauth1 body', async () => {
  const [main] = await convertFile('test/material/2/oauth1-body.json');
  expect(main).toMatchSnapshot();
});

test('oauth1 address', async () => {
  const [main] = await convertFile('test/material/2/oauth1-address.json');
  expect(main).toMatchSnapshot();
});

test('oauth2 header', async () => {
  const [main] = await convertFile('test/material/2/oauth2-header.json');
  expect(main).toMatchSnapshot();
});

test('oauth2 address', async () => {
  const [main] = await convertFile('test/material/2/oauth2-address.json');
  expect(main).toMatchSnapshot();
});

test('inherit collection', async () => {
  const [main] = await convertFile('test/material/2/inherit-collection.json');
  expect(main).toMatchSnapshot();
});

test('inherit folder', async () => {
  const [main] = await convertFile('test/material/2/inherit-folder.json');
  expect(main).toMatchSnapshot();
});

test('inherit nested', async () => {
  const [main] = await convertFile('test/material/2/inherit-nested.json');
  expect(main).toMatchSnapshot();
});

test('apikey', async () => {
  const [main] = await convertFile('test/material/2/apikey.json');
  expect(main).toMatchSnapshot();
});
