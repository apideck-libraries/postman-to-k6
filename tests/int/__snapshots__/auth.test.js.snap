// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apikey 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      config.headers["Authorization"] = "secretApiKey";
    }
  });
}
"
`;

exports[`awsv4 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/spo-gpo.js";
import "./libs/shim/core.js";
import URI from "./libs/urijs.js";
import aws4 from "./libs/aws4.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const address = new URI(config.address);
      const options = {
        method: "GET",
        protocol: address.protocol(),
        hostname: address.hostname(),
        port: address.port(),
        path: address.path() + address.search(),
        body: config.data,
        region: "region",
        service: "service"
      };
      const credential = {
        accessKeyId: "accesskey",
        secretAccessKey: "secretkey",
        sessionToken: "session"
      };
      const signed = aws4.sign(options, credential);
      const [path, query = ""] = signed.path.split("?");
      config.address = new URI()
        .protocol(address.protocol())
        .hostname(signed.hostname)
        .path(path)
        .query(query)
        .toString();
      Object.assign(config.headers, signed.headers);
    }
  });
}
"
`;

exports[`basic 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import URI from "./libs/urijs.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const address = new URI(config.address);
      address.username("user123");
      address.password("secret");
      config.address = address.toString();
      config.options.auth = "basic";
    }
  });
}
"
`;

exports[`bearer 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      config.headers.Authorization = "Bearer secrettoken";
    }
  });
}
"
`;

exports[`digest 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import URI from "./libs/urijs.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const address = new URI(config.address);
      address.username("user123");
      address.password("secret");
      config.address = address.toString();
      config.options.auth = "digest";
    }
  });
}
"
`;

exports[`inherit collection 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      config.headers.Authorization = "Bearer token";
    }
  });
}
"
`;

exports[`inherit folder 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/",
      auth(config, Var) {
        config.headers.Authorization = "Bearer token";
      }
    });
  });
}
"
`;

exports[`inherit nested 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder1", function() {
    group("TestFolder2", function() {
      group("TestFolder3", function() {
        group("TestFolder4", function() {
          postman[Request]({
            name: "TestRequest",
            method: "GET",
            address: "http://example.com/",
            auth(config, Var) {
              config.headers.Authorization = "Bearer token";
            }
          });
        });
      });
    });
  });
}
"
`;

exports[`noauth 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`ntlm 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import URI from "./libs/urijs.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const address = new URI(config.address);
      address.username("user123");
      address.password("secret");
      config.address = address.toString();
      config.options.auth = "ntlm";
    }
  });
}
"
`;

exports[`oauth1 address 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import OAuth from "./libs/oauth-1.0a.js";
import URI from "./libs/urijs.js";
import { hmac } from "k6/crypto";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    data: "line1\\nline2\\nline3\\n",
    auth(config, Var) {
      const options = {
        consumer: {
          key: "conkey",
          secret: "consec"
        },
        signature_method: "HMAC-SHA1",
        hash_function(data, key) {
          return hmac("sha1", key, data, "base64");
        },
        version: "1.0",
        realm: "realm@example.com"
      };
      const request = {
        method: config.method,
        url: config.address,
        data: {
          oauth_timestamp: "1",
          oauth_nonce: "10"
        }
      };
      const token = {
        key: "tokkey",
        secret: "toksec"
      };
      const oauth = OAuth(options);
      const auth = oauth.authorize(request, token);
      const address = new URI(config.address);
      for (const key of Object.keys(auth)) {
        address.addQuery(key, auth[key]);
      }
      config.address = address.toString();
    }
  });
}
"
`;

exports[`oauth1 body 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import OAuth from "./libs/oauth-1.0a.js";
import { hmac } from "k6/crypto";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "POST",
    address: "http://example.com/",
    auth(config, Var) {
      const options = {
        consumer: {
          key: "conkey",
          secret: "consec"
        },
        signature_method: "HMAC-SHA1",
        hash_function(data, key) {
          return hmac("sha1", key, data, "base64");
        },
        version: "1.0",
        realm: "realm@example.com"
      };
      const request = {
        method: config.method,
        url: config.address,
        data: {
          oauth_timestamp: "1",
          oauth_nonce: "10"
        }
      };
      const token = {
        key: "tokkey",
        secret: "toksec"
      };
      const oauth = OAuth(options);
      const auth = oauth.authorize(request, token);
      Object.assign(config.data, auth);
    }
  });
}
"
`;

exports[`oauth1 header sha1 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import OAuth from "./libs/oauth-1.0a.js";
import { hmac } from "k6/crypto";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const options = {
        consumer: {
          key: "conkey",
          secret: "consec"
        },
        signature_method: "HMAC-SHA1",
        hash_function(data, key) {
          return hmac("sha1", key, data, "base64");
        },
        version: "1.0",
        realm: "realm@example.com"
      };
      const request = {
        method: config.method,
        url: config.address,
        data: {
          oauth_timestamp: "1",
          oauth_nonce: "10"
        }
      };
      const token = {
        key: "tokkey",
        secret: "toksec"
      };
      const oauth = OAuth(options);
      const auth = oauth.toHeader(oauth.authorize(request, token));
      Object.assign(config.headers, auth);
    }
  });
}
"
`;

exports[`oauth1 header sha256 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import OAuth from "./libs/oauth-1.0a.js";
import { hmac } from "k6/crypto";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const options = {
        consumer: {
          key: "conkey",
          secret: "consec"
        },
        signature_method: "HMAC-SHA256",
        hash_function(data, key) {
          return hmac("sha256", key, data, "base64");
        },
        version: "1.0",
        realm: "realm@example.com"
      };
      const request = {
        method: config.method,
        url: config.address,
        data: {
          oauth_timestamp: "1",
          oauth_nonce: "10"
        }
      };
      const token = {
        key: "tokkey",
        secret: "toksec"
      };
      const oauth = OAuth(options);
      const auth = oauth.toHeader(oauth.authorize(request, token));
      Object.assign(config.headers, auth);
    }
  });
}
"
`;

exports[`oauth1 header text 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import OAuth from "./libs/oauth-1.0a.js";
import { hmac } from "k6/crypto";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const options = {
        consumer: {
          key: "conkey",
          secret: "consec"
        },
        signature_method: "PLAINTEXT",
        version: "1.0",
        realm: "realm@example.com"
      };
      const request = {
        method: config.method,
        url: config.address,
        data: {
          oauth_timestamp: "1",
          oauth_nonce: "10"
        }
      };
      const token = {
        key: "tokkey",
        secret: "toksec"
      };
      const oauth = OAuth(options);
      const auth = oauth.toHeader(oauth.authorize(request, token));
      Object.assign(config.headers, auth);
    }
  });
}
"
`;

exports[`oauth2 address 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import URI from "./libs/urijs.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      const address = new URI(config.address);
      address.addQuery("access_token", "token");
      config.address = address.toString();
    }
  });
}
"
`;

exports[`oauth2 header 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    auth(config, Var) {
      config.headers.Authorization = "Bearer token";
    }
  });
}
"
`;
